### PR TITLE
feat: OpenRPC

### PR DESCRIPTION
**Pull request by crypto-facs**
Closes: #XXX

## Description

First PR to support OpenRPC discover method. I had to remove the `trimpath` from the build in order for this to work but open to discuss that

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -94,10 +94,6 @@ ldflags += $(LDFLAGS)
 ldflags := $(strip $(ldflags))
 
 BUILD_FLAGS := -tags "$(build_tags)" -ldflags '$(ldflags)'
-# check for nostrip option
-ifeq (,$(findstring nostrip,$(COSMOS_BUILD_OPTIONS)))
-  BUILD_FLAGS += -trimpath
```
Comment (yihuang) [inline Makefile:99]: I think trimpath is for reproducible build?

Comment (crypto-facs) [reply Makefile:99]: yeah.. this part is the one I wanted to discuss as I had put on my comment. Having the `trimpath` breaks the reflection module we are using for openRPC but it does help for reproducible builds

Comment (fedekunze) [reply Makefile:99]: @yihuang @crypto-facs is this currently blocking this PR from getting merged?

**Review by fedekunze — COMMENTED** [review]
LGTM, pending @yihuang's comment. Can you also add a new feature changelog entry?

**Comment by fedekunze** [comment]
@crypto-facs what's the status of this PR?

**Pull request closed without merging**